libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/ClientManager.java

```diff
@@ -790,6 +801,19 @@ public LoginOptions(String loginUrl, String passcodeHash, String oauthCallbackUr
             this.setJwt(jwt);
         }
 
+        public LoginOptions(String loginUrl, String passcodeHash, String oauthCallbackUrl,
+                            String oauthClientId, String[] oauthScopes, String clientSecret, String jwt,
+                            HashMap<String,String> additionalParameters) {
+            this(loginUrl, passcodeHash, oauthCallbackUrl, oauthClientId, oauthScopes,clientSecret,jwt);
+            this.setAdditionalParameters(additionalParameters);
+        }
+
+        public void setAdditionalParameters(HashMap<String,String> additionalParameters) {
+            this.additionalParameters = additionalParameters;
+            bundle.putSerializable(AuthenticatorService.KEY_ADDL_PARAMS,additionalParameters);
```

Comment:
Tried to be consistent with the definition of this already designed class. Clearly this problem exists with all the class members. Will look into the overall design and see if I can clean this class up. I don't see why the bundle cannot be created on demand(asBundle).